worker/uniter/runner/jujuc/network-get.go

```diff
@@ -161,3 +166,27 @@ func (c *NetworkGetCommand) Run(ctx *cmd.Context) error {
 	}
 	return c.out.Write(ctx, keyValues)
 }
+
+// This addresses the immediate problem of
+// https://bugs.launchpad.net/juju/+bug/1721368, but the hostname can populate
+// both the egress subnet cidr and the ingress addreses. These too should be
+// resolved. In addition these values probably should not be stored as hostnames
+// but rather the IP, that is, it might be better to do the resolution on input
+// rather than output (network-get) as we do here.
+func resolveNetworkInfoAddresses(networkInfoResult params.NetworkInfoResult) params.NetworkInfoResult {
+	logger.Debugf("Resolving Addresses")
+	for i, networkInfo := range networkInfoResult.Info {
+		for j, interfaceAddress := range networkInfo.Addresses {
+			if ip := net.ParseIP(interfaceAddress.Address); ip != nil {
+				continue
+			}
+			resolvedAddress, err := net.LookupHost(interfaceAddress.Address)
+			if err != nil {
+				logger.Warningf("The address %q is neither an IP address or a resolvable hostname", interfaceAddress.Address)
```

Comment:
should this include the actual error, so it's not lost

Reply:
Yes, it should. Will do.